### DISCUSSION TITLE
What AlphaPixel has been up to

### DISCUSSION THREAD

**Discussion by XenonofArcticus** [openscenegraph/OpenSceneGraph#1235] [General]
Hey all. Been away from communications for a while on some heavy projects lately, but coming up for air now and wanted to generally mention some of the interesting notable stuff we've worked on in the space. Most of our clients are confidential and closed-source applications, but I can vaguely describe the technologies we've been working with in the OSG/VSG/OGL/Vulkan/osgEarth/etc space.

We are seeing interest in Vulkan and VSG (and osgEarth's successors vsgEarth and Rocky) but only exploratory work from clients on them currently. 

We continue to do work on OSG-based projects, but try to convince new greenfield developments to go with VSG. Our customers are hesitant to leave the comforts of OSG still, but we think we're making progress.

We have done a TON of air and space-domain visualization work, often involving osgEarth, but also seeing Unreal Engine+Cesium. The new Google photorealistic 3D Tiles are very interesting for rapidly creating a complete Earth model with detailed scenery basically instantly. I know the Google 3D tiles have been tested in Cesium and Cesium for Unreal (not sure about other Cesium environments) and Pelican Mapping has shown me their successful tests of Google's 3D tiles in osgEarth, so that's very exciting. Deploying a fully-built-out 3D globe is now easier than ever before.

We have also done a lot of VR/AR/MR/xR/whateveR and multiview rendering lately. We did some extensive work on high-end headsets like Varjo, HoloLens 2, Xtal and Pimax, with support for Foveated Rendering and high framerate, low-latency rendering in both OSG and Unreal Engine environments. Frequently, applications trying to hit 120Hz, low-latency quad-view (stereo with foveated) need a thorough optimizing to be able to keep up with the rendering pipeline.

We have new customers for our native C++ NATO MIL-STD-2525 symbology toolkit, which is obviously used primarily in the military training sector. 

We are doing a lot of new stuff with AI and Deep/Machine Learning. Have been using Stable Diffusion for processing basic 3d renderings produced by polygonal renderers like OSG into photorealistic stills. The pipeline is still very slow (15 seconds per frame at 512x512 pixels) and is tricky to control for temporal coherency (animations instead of just stills) but is very promising. I don't know if it will achieve realtime simulation speeds and resolutions anytime soon, but I can see having enough ML compute power in 5 years that it could be practical.

We've also done a lot of computer vision/navigation-related work, especially involving OpenCV and autonomous/robotic vehicles and spacecraft. We used OSG and osgEarth to generate synthetic visual imagery to train a vision navigation system for GPS-denied environments. We are experimenting with using Stable Diffusion to generate synthetic training data for vision systems.

We've worked on some realtime low-latency h264 systems, for both edge-computing rendering of 3d content on embedded (like NVidia Jetson Orin and friends) systems for transmission of xR media over next-generation wireless networks. We've also used OSG to render HUDs showing metadata and overlays onto live video captured from SDI digital video cameras and streamed to an onboard digital video network using h264. This one was especially critical to be low-latency. We've done other embedded systems from Raspberry Pi to some of the new RISC-V systems, which generally have VERY limited GPUs compared to the Jetson's nice powerful NVidia GPU.

That's all the fun stuff I can think of right now. We have several OSG and osgEarth (and Vulken, et al) experienced developers now, and some space in our schedule the rest of 2023, so if any of the above sounds like things you expect to be doing, feel free to reach out to us. We're always happy to provide advice and insight gratis, even if we're not working on a project for you for money.

If this has been interesting to you, maybe reply and post something about what YOU have been up to!

**Comment by andek714** [comment]
Hi all,

Thanks for sharing Chris! Really interesting to read about your continued adventures in the wonderful world of visualization. I thought I would give an update of what we have been up to at Remograph lately as well. It won't be as comprehensive or varied, but hopefully somewhat interesting anyway.

Remo 3D continues to evolve and is still a valued modeling tool for many customers worldwide, not least for its native OpenFlight support and specific focus on realtime features. Examples of recent improvements are boolean tools, conversion to geocentric coordinates and detailed control of light sources.

In April we finally released our new terrain generating cloud service called Remoscape. Just like Remo 3D, of course its totally based on OpenSceneGraph under the hood, but the number of export formats will continue to grow. As a user-friendly and simple alternative to many other competent tools out there, it has quickly attracted quite some interest in the simulation community. However, with many of the presumptive customers in the defense industry, we have concluded that (unsurprisingly) we need to prioritize providing a locally installed version as well. Together with new export formats (e.g. 3D tiles, CDB), more details in the produced landscapes and connecting to online data source services like e.g. OpenStreetMap, this will be our focus in the coming months.

For both Remo 3D and Remoscape, we have had a great technical collaboration with MAK Technologies, making our products interoperable. For example, there is a specific MAK export format in Remoscape. With both companies building their products upon OpenSceneGraph, we clearly share a common technical understanding and lingo.

Personally, I have been very involved in Spotscale as well, where I'm responsible for visualizing their ultra-high detail building models produced from drone imagery using photogrammetry. Of course by using OpenSceneGraph there as well, and OSGJS which is still going strong despite officially being discontinued.

As always, there is too much fun to fill one's time with and I wish there were 48 hours in a day. I'd love to hear from more fellow OSGers what you have been up to lately as well!

/Andreas

**Reply by XenonofArcticus** [reply]
FWIW, yes, I'd totally agree on the need for both local and cloud 3D Tiles. 

We actually did some work with/for MAK on their MAK One products, including VR-Vantage and VR-Engage, but I'm not able to comment on what specifically we did on those products. It's exciting to see them working with your terrain data services and 3D tiles, and they're a great product.
